src/backend/commands/view.c

```diff
@@ -71,6 +72,27 @@ DefineVirtualRelation(RangeVar *relation, List *tlist, bool replace,
 											exprType((Node *) tle->expr),
 											exprTypmod((Node *) tle->expr),
 											exprCollation((Node *) tle->expr));
+			/*
+			 * If the dialect is TSQL then preserve the constraints.
+			 */
+			if (sql_dialect == SQL_DIALECT_TSQL)
```

Comment:
It should be a hook. Second, why should a view needs to inherit properties of columns which are relevant from insert perspective?